inspect_byproducts.py

```diff
@@ -0,0 +1,176 @@
+        1 - if the inspection is unsuccessful
+        2 - if the user supplied arguments are invalid
+
+
+
+    """
+    if not os.path.exists(link):
+        print("The path to the link file is invalid")
+        return 2
+
+    else:
+      l = link_import.read_from_file(link)
+
+      try:
+        std_out_err = l.byproducts[std]
+      except KeyError:
+        raise KeyError("The field corresponding to " + std + " in the link "
+                                                             "file is empty.")
+
+      switch_dict = {'is': (std_out_err == inputstring),
+                     'is not': (std_out_err != inputstring),
+                     'contains': (std_out_err.find(inputstring)),
+                     'contains not': (std_out_err.find(inputstring))}
+
+      if operator == 'is':
+        if switch_dict[operator]:
+          return 0
+
+      elif operator == 'is not':
+        if switch_dict[operator]:
+          return 0
+
+      elif operator == 'contains':
+        if switch_dict[operator] != -1:
+          return 0
+
+      elif operator == 'contains not':
+        if switch_dict[operator] == -1:
+          return 0
+
+
+      return 1
+
+
+
+def main():
+    """Parse the arguments and call inspect_byproducts. """
+    parser = argparse.ArgumentParser(
+        description="Inspects the byproducts of a step")
+
+    lpad = (len(parser.prog) + 1) * " "
+
+    parser.usage = ("\n"
+                    "%(prog)s --link <path to link metadata>\n{0}"
+                    "[--stdout | --stderr]\n{0}"
+                    "[--is | --is-not | --contains | --contains-not]\n{0}"
+                    "string\n{0}"
+                    "[--verbose]\n\n"
+                    .format(lpad))
+
+    in_toto_args = parser.add_argument_group("in-toto-inspection options")
+
+    in_toto_args.add_argument("-l", "--link", type=str, required=True,
+                              help="Link metadata file to use for inspection "
+                                  "of the step")
+
+    in_toto_args.add_argument("-st", "--outerr",
+                              type=str, required=True, help="when stdout or "
+                                  "stderr is a byproduct")
+
+    in_toto_args.add_argument("-o", "--operator",
+                              type=str, required=True, help="whether the "
+                                  "stdout or stderr is, is not, \ "
+                                  "contains, contains not, the input string")
+
+    in_toto_args.add_argument("string", type=str,
+                              help="The string to which the return value "
+                                  "should be compared")
+
+    in_toto_args.add_argument("-v", "--verbose", dest="verbose",
+                              help="Verbose execution.", default=False,
+                                  action="store_true")
+
+    args = parser.parse_args()
+    args.operator = args.operator.lower()
+    args.outerr = args.outerr.lower()
```

Comment:
As I said, nothing against it, the only reason why I haven't done it so far is that main() pretty much does not do anything else but argparsing. But let's not waste time on this. Please sachit go ahead and move the argparsing to an extra function.